test/functional/hostgroups_controller_test.rb

```diff
@@ -136,4 +136,34 @@ def setup_user(operation, type = 'hostgroups')
     assert_response :success
     assert_template :partial => "common_parameters/_inherited_parameters"
   end
+
+  describe "parent attributes" do
+    before do
+      @base = FactoryGirl.create(:hostgroup)
+      @base.group_parameters << GroupParameter.create(:name => "x", :value => "original")
+    end
+
+    it "creates a hostgroup with a parent parameter" do
+      post :create, {"hostgroup" => {"name"=>"test_it", "parent_id" => @base.id, :realm_id => realms(:myrealm).id}, "parent_parameter_0_key"=>"x", "parent_parameter_0_value"=>"overridden"}, set_session_user
+      assert_redirected_to hostgroups_url
+      hostgroup = Hostgroup.where(:name => "test_it").last
+      assert_equal hostgroup.parameters["x"], "overridden"
+    end
+
+    it "updates a hostgroup with a parent parameter" do
+      child = FactoryGirl.create(:hostgroup, :parent => @base)
+      assert child.parameters["x"] == "original"
+      post :update, {"id" => child.id, "hostgroup" => {"name" => child.name}, "parent_parameter_0_key"=>"x", "parent_parameter_0_value"=>"overridden" }, set_session_user
+      assert_redirected_to hostgroups_url
+      assert_equal child.parameters["x"], "overridden"
+    end
+
+    it "updates a hostgroup with a parent parameter without updating the parameter" do
+      child = FactoryGirl.create(:hostgroup, :parent => @base)
+      assert child.parameters["x"] == "original"
+      post :update, {"id" => child.id, "hostgroup" => {"name" => "new_name"} }, set_session_user
```

Comment:
I think the parameters need updating here to show the latest bug that was fixed, as it's not representative of the UI.  The UI is always going to send fields for a parent parameters (since they're always in the form), but with an empty value.